executor/update.go

```diff
@@ -176,35 +153,34 @@ func (e *UpdateExec) fetchChunkRows(ctx context.Context) error {
 		composeFunc = e.composeNewRow
 	}
 	memUsageOfChk := int64(0)
+	totalNumRows := 0
 	for {
 		e.memTracker.Consume(-memUsageOfChk)
 		err := Next(ctx, e.children[0], chk)
 		if err != nil {
-			return err
+			return 0, err
 		}
 
 		if chk.NumRows() == 0 {
 			break
 		}
 		memUsageOfChk = chk.MemoryUsage()
 		e.memTracker.Consume(memUsageOfChk)
-		firstRowIdx := globalRowIdx
 		for rowIdx := 0; rowIdx < chk.NumRows(); rowIdx++ {
 			chunkRow := chk.GetRow(rowIdx)
 			datumRow := chunkRow.GetDatumRow(fields)
 			newRow, err1 := composeFunc(globalRowIdx, datumRow, colsInfo)
 			if err1 != nil {
-				return err1
+				return 0, err1
+			}
+			if err := e.exec(ctx, e.children[0].Schema(), datumRow, newRow); err != nil {
+				return 0, err
 			}
-			e.rows = append(e.rows, datumRow)
-			e.newRowsData = append(e.newRowsData, newRow)
-			globalRowIdx++
 		}
-		e.memTracker.Consume(types.EstimatedMemUsage(e.rows[firstRowIdx], globalRowIdx-firstRowIdx))
-		e.memTracker.Consume(types.EstimatedMemUsage(e.newRowsData[firstRowIdx], globalRowIdx-firstRowIdx))
```

Comment:
Why memory tracker is removed?

Reply:
I think we needn't count them now. They are not saved a second time.